ufl/measure.py

```diff
@@ -32,6 +32,8 @@
     # === Integration over topological dimension - 1:
     ("exterior_facet", "ds"),  # Over one-sided exterior facets of a mesh
     ("interior_facet", "dS"),  # Over two-sided facets between pairs of adjacent cells of a mesh
+    # === Integration over topological dimension - 2:
+    ("edge", "dl"),  # Over edges of a mesh
```

Comment:
edges are only tdim - 2 in a 3d mesh. Should these be called "ridges" instead (See eg https://en.wikipedia.org/wiki/Face_(geometry)#k-face)

Reply:
Good catch! Presumably `dl` integrates over points in 2d (just like `ds` and `dS` in 1D).

Reply:
> ridge

I was thinking about this when looking through UFL. I guess we should decide if we want to support "Edge" (i.e. only 3D-1D), or ridge (2D-0D, 3D-1D)? @cdaversin do you have any opinions?

Reply:
I think this is conflating "things that are expressible in UFL" with "things that are supported in Dolfinx". Lots of things can be expressed in UFL that individual solvers don't support. The point is that the UFL notation should match the maths as closely as possible so that we don't have to go back and make a backward incompatible change when someone expands their capabilities in the future.

I think that the UFL feature should be codimension 2, even if that's not what is supported in Dolfinx right now.